### PR TITLE
sync: Extra properties update

### PR DESCRIPTION
**Pull request by artem-lunarg**
* Sort extra properties
* Add option to pretty print extra properties.

### CODE REVIEW

File: layers/VkLayer_khronos_validation.json.in [hunk 1]
```diff
@@ -436,6 +436,20 @@
                                                     { "key": "validate_sync", "value": true }
                                                 ]
                                             }
```
Comment (spencer-lunarg) [inline layers/VkLayer_khronos_validation.json.in:438]: I think you can nest this inside here

```json
"key": "syncval_message_extra_properties",
"label": "Extra key-value properties",
"description": "...",
"description": "...",
"type": "BOOL",
"default": false,
"dependence": {
    "mode": "ALL",
    "settings": [
        { "key": "validate_sync", "value": true }
    ]
},
"settings": [
    {
        "key": "syncval_message_pretty_print_extra_properties",
        "label": "Pretty print extra properties",
        ....
    }
]
```

File: layers/VkLayer_khronos_validation.json.in [hunk 2]
```diff
@@ -436,6 +436,20 @@
...
+                                        },
+                                        {
+                                            "key": "syncval_message_pretty_print_extra_properties",
```
Comment (spencer-lunarg) [inline layers/VkLayer_khronos_validation.json.in:441]:
```suggestion
                                            "key": "syncval_message_extra_properties_pretty_print",
```

?

**Review by spencer-lunarg — APPROVED** [review]
questions about the JSON, which can always be sorted out between now and the next SDK if needed

File: layers/VkLayer_khronos_validation.json.in (hunk 2 above)
Comment (artem-lunarg) [reply layers/VkLayer_khronos_validation.json.in:441]: Yes, I considered this for a moment, was not sure, will update. I guess that's a choice between more consistent structure and using regular english. On the other side putting "message" on the first place already made a choice to prefer regular structure over english.

Comment (spencer-lunarg) [reply layers/VkLayer_khronos_validation.json.in:441]: If people are using vkconfig, this `key` name means nothing, so I view its english as not being the most important (granted my english is the worst on the team to begin with...)

File: layers/VkLayer_khronos_validation.json.in (hunk 1 above)
Comment (artem-lunarg) [reply layers/VkLayer_khronos_validation.json.in:438]: Not sure. Linear structure matches linear strucure in the UI (one checkbox after another). Under other equal conditions nesting two blocks in json makes final structure quite complex.

Comment (artem-lunarg) [reply layers/VkLayer_khronos_validation.json.in:438]: image

**Pull request merged**